tests/functional/LocaleTest.php

```diff
@@ -11,6 +11,41 @@
 class LocaleTest extends CommandUnishTestCase
 {
 
+    public function testLocaleExport()
+    {
+        $this->setUpDrupal(1, true);
+        $this->drush('pm:enable', ['language', 'locale']);
+        $this->drush('language:add', ['nl'], ['skip-translations' => null]);
+
+        $source = Path::join(__DIR__, '/resources/devel.nl.po');
+
+        $this->drush('locale:import', ['nl', $source]);
+        $this->assertTranslation('Devel', 'NL Devel', 'nl', 0);
+
+        // Export standard translations.
+        $this->drush('locale:export', ['nl'], ['types' => 'not-customized']);
+        $this->assertGettextTranslation('Devel', 'NL Devel');
+
+        // Export customized translations.
+        $this->drush('sql:query', ["UPDATE locales_target SET translation = 'CUSTOM Devel', customized = 1"]);
+        $this->assertTranslation('Devel', 'CUSTOM Devel', 'nl', 1);
+        $this->drush('locale:export', ['nl'], ['types' => 'customized']);
+
+        // Export untranslated strings.
+        $this->drush('sql:query', ["INSERT locales_source SET source = 'Something untranslated'"]);
+        $this->drush('locale:export', ['nl'], ['types' => 'not-translated']);
+
+        // Export all.
+        $this->drush('locale:export', ['nl']);
+        $this->assertGettextTranslation('Devel', 'CUSTOM Devel');
+        $this->assertGettextTranslation('Something untranslated', '');
+
+        // Export template file.
+        $this->drush('locale:export', ['nl'], ['template' => null]);
+        $this->assertGettextTranslation('Devel', '');
+        $this->assertGettextTranslation('Something untranslated', '');
+    }
+
     public function testLocaleImport()
     {
         $this->setUpDrupal(1, true);
```

Comment:
We should do 1 setUpDrupal() per class. See other tests for examples ... Or even better, see if this can be an Integration test instead of a Functional test.

Reply:
@weitzman, can you explain which groups of tests you refer to with "Integration test"? Which test(s) can serve as an example?

Reply:
See files in https://github.com/drush-ops/drush/tree/master/tests/integration